frontend/include/chpl/framework/serialize-functions.h

```diff
@@ -49,7 +51,11 @@ template<typename T> struct deserialize;
 
 class Serializer {
 private:
+  using stringCacheType = std::map<const char*, std::pair<int, size_t>>;
```

Comment:
Might be good to comment where the memory for the string pool lives and whether or not the strings are expected to be null terminated.